test/protocol-tests-core/src/main/java/software/amazon/awssdk/protocol/asserts/unmarshalling/UnmarshalledResultAssertion.java

```diff
@@ -62,7 +71,11 @@ private void assertFieldEquals(Field field, Object actual, Object expectedResult
             assertTrue(IOUtils.contentEquals((InputStream) field.get(expectedResult),
                                              (InputStream) field.get(actual)));
         } else {
-            assertReflectionEquals(field.get(expectedResult), field.get(actual));
+            Difference difference = CustomComparatorFactory.getComparator()
```

Comment:
Note to reviewer: this fixes an issue where the comparator is using reflection to compare instants (which fails on JDK 17), because our comparator library is old enough that it doesn't support Instant types. (It hasn't been updated since 2017).